src/Keboola/DbExtractor/Extractor/Firebird.php

```diff
@@ -53,4 +57,26 @@ public function simpleQuery(array $table, array $columns = array()): string
     {
         throw new UserException('This component does not yet support simple queries');
     }
+
+    private function runRetriableQuery(string $query, string $errorMessage = ''): array
+    {
+        $retryProxy = new DbRetryProxy(
+            $this->logger,
+            DbRetryProxy::DEFAULT_MAX_TRIES,
+            [\PDOException::class, \ErrorException::class, \Throwable::class]
+        );
+        return $retryProxy->call(function () use ($query, $errorMessage): array {
+            try {
+                $stmt = $this->db->prepare($query);
+                $stmt->execute();
+                return $stmt->fetchAll(\PDO::FETCH_ASSOC);
+            } catch (\Throwable $e) {
+                throw new UserException(
+                    $errorMessage . ': ' . $e->getMessage(),
+                    0,
+                    $e
+                );
```

Comment:
Same comment here like I made on the Impala one 
https://github.com/keboola/db-extractor-impala/pull/32#discussion_r353698571

Reply:
ok, this is imho better